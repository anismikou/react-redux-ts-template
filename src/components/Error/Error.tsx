import React, { FC } from 'react'
import styled from 'styled-components'

interface IErrorProps {
    className?: string
    title: string
    children: string
}

const Error: FC<IErrorProps> = ({ className, title, children }) => (
    <div className={className}>
        <h1>{title}</h1>
        <p>{children}</p>
    </div>
)

const StyledError = styled(Error)`
    color: red;
`

export default StyledError
