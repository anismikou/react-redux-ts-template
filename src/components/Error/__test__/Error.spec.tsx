import { shallow } from 'enzyme'
import React from 'react'
import Error from '../Error'

describe('<Error />', () => {
    test('should render correctly', () => {
        const wrapper = shallow(<Error title="title">Message</Error>)
        expect(wrapper).toMatchSnapshot()
    })
})
