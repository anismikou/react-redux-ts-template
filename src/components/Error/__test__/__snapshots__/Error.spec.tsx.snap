// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should render correctly 1`] = `
<Error
  className="sc-AxjAm kokBMW"
  title="title"
>
  Message
</Error>
`;
